### DISCUSSION TITLE
Stack update fails due to missing permission iam:GetRole

### DISCUSSION THREAD

**Discussion by vvdt** [mozilla/hubs#4687] [Hubs Cloud Questions]
Hey, 

I'm seeing more failed Stack updates, with as reason 'AccessDenied. User doesn't have permission to call iam:GetRole'  on some services. 

![Screenshot 2021-09-29 at 13 14 53](https://user-images.githubusercontent.com/3598257/135271725-930caefc-aee7-4fe7-99e3-31b3f1abfa71.png)

I tried to make sure my user that deploys the stack has all IAM permissions but no luck so far. Btw, I did not assign a custom IAM role to the stack, so I assume it takes the permissions of the IAM user that is running the stack.

I can't find much about the required permissions here or in Discord, so I'm a bit stuck in how to resolve this. Is there some information about hubs' permissions that I missed?

**Comment by pattersonbl2** [comment]
Are you able to Dm me in the discord your IAM user permissions?   This error occurs when the IAM Users don't have the correct permissions to deploy the CF.  Check your user or a user group to see if you have iam:passRole assigned.  Our stack assumes have full admin rights over your ASW account.

**Reply by vvdt** [reply]
Thanks for the quick reply. I can't find your Discord name but emailed your mine on your github email. The IAM user has policies IAMFullAccess and AdministratorAccess (in that order), which both seem to include iam:passRole and iam:getRole.

**Reply by Ulriichde1st** [reply]
@Pattersonbl2 @vvdt Was this issue solved if I may ask? I'm currently facing the same.

**Comment by Ritika-Nunia** [comment]
I am facing the same issue, even the IAM user has policies IAMFullAccess and AdministratorAccess include iam:passRole and iam:getRole. What I need to do? Any suggestions

**Reply by Rudy-rumbleon** [reply]
I am facing the same issue while using serverless deploy. Though user have all lambda execution permissions the cloudformation stack is returning error UPDATE_FAILED: FetchAllStoresForOpportunitiesLambdaFunction (AWS::Lambda::Function)
AccessDenied. User doesn't have permission to call iam:GetRole but the user has all the lambda execution permissions
